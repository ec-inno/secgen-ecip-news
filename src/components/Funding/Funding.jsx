import React from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import FileDownload from '../FileDownload';

const Funding = ({ funding }) => {
  if (!funding) return '';

  const hasSponsors = funding.sponsors && funding.sponsors.length !== 0;
  const hasTotalAmount = Boolean(
    funding.totalAmount && Number(funding.totalAmount) !== 0
  );

  // Double-check if's worth showing this section.
  if (
    !funding.lastUpdate &&
    !funding.document &&
    !hasSponsors &&
    !hasTotalAmount
  ) {
    return '';
  }

  const { t } = useTranslation();

  return (
    <>
      <h2 className="ecl-u-type-heading-2">
        {t('Sources of support and funding')}
      </h2>
      {funding.lastUpdate && (
        <p className="ecl-u-type-paragraph">
          {t('Last update')}
          {': '}
          {funding.lastUpdate}
        </p>
      )}

      {hasSponsors && (
        <table className="eci-table ecl-u-type-paragraph">
          <tbody>
            <tr>
              <th>{t('Name of sponsor')}</th>
              <th>
                {t('Date')}
                <sup>[1]</sup>
              </th>
              <th>
                {t('Amount in EUR')}
                <sup>[2]</sup>
              </th>
            </tr>

            {funding.sponsors.map((sponsor, key) => (
              <tr className={key % 2 === 1 ? 'dark-row' : ''} key={key}>
                <td>{sponsor.name}</td>
                <td>{sponsor.date}</td>
                <td>{sponsor.amount}</td>
              </tr>
            ))}
          </tbody>
        </table>
      )}
      {hasTotalAmount && (
        <p className="ecl-u-type-paragraph ecl-u-type-bold">
          {t('Total amount of support and funding')}
          {': '}â‚¬{funding.totalAmount}
        </p>
      )}
      {hasSponsors && (
        <p className="ecl-u-type-paragraph-s">
          [1]{' '}
          {t(
            'Date when the contribution was received. In case of in kind contributions, date when the in kind contribution started.'
          )}
          <br />
          [2]{' '}
          {t(
            'Actual amount of funding provided or estimated value of in kind contribution.'
          )}
        </p>
      )}
      {funding.document && (
        <p className="ecl-u-type-paragraph ecl-u-type-bold">
          {t('Further information on support and funding')}:
          <FileDownload file={funding.document} />
        </p>
      )}
    </>
  );
};

Funding.propTypes = {
  funding: PropTypes.shape({
    lastUpdate: PropTypes.string,
    totalAmount: PropTypes.number,
    sponsors: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        date: PropTypes.string,
        amount: PropTypes.number,
      })
    ),
  }),
};

export default Funding;
