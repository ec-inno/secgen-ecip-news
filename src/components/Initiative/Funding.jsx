import React from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import Document from './Document';

const Funding = ({ funding }) => {
  if (!funding) return '';

  const { t } = useTranslation();

  return (
    <>
      <h2 className="ecl-u-type-heading-2">
        {t('Sources of support and funding')}
      </h2>
      {funding.lastUpdate && (
        <p className="ecl-u-type-paragraph">
          {t('Last update')}
          {': '}
          {funding.lastUpdate}
        </p>
      )}

      <table className="eci-table ecl-u-type-paragraph">
        <tbody>
          <tr>
            <th>{t('Name of sponsor')}</th>
            <th>
              {t('Date')}
              <sup>[1]</sup>
            </th>
            <th>
              {t('Amount in EUR')}
              <sup>[2]</sup>
            </th>
          </tr>
          {funding.sponsors
            ? funding.sponsors.map((sponsor, key) => (
                <tr className={key % 2 === 1 ? 'dark-row' : ''} key={key}>
                  <td>{sponsor.name}</td>
                  <td>{sponsor.date}</td>
                  <td>{sponsor.amount}</td>
                </tr>
              ))
            : ''}
        </tbody>
      </table>
      {funding.totalAmount && (
        <p className="ecl-u-type-paragraph ecl-u-type-bold">
          {t('Total amount of support and funding')}
          {': '}â‚¬{funding.totalAmount}
        </p>
      )}
      <p className="ecl-u-type-paragraph-s">
        [1]{' '}
        {t(
          'Date when the contribution was received. In case of in kind contributions, date when the in kind contribution started.'
        )}
        <br />
        [2]{' '}
        {t(
          'Actual amount of funding provided or estimated value of in kind contribution.'
        )}
      </p>
      {funding.document && (
        <p className="ecl-u-type-paragraph ecl-u-type-bold">
          {t('Further information on support and funding')}:
          <Document file={funding.document} />
        </p>
      )}
    </>
  );
};

Funding.propTypes = {
  funding: PropTypes.shape({
    lastUpdate: PropTypes.string,
    totalAmount: PropTypes.number,
    sponsors: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        date: PropTypes.string,
        amount: PropTypes.number,
      })
    ),
  }),
};

export default Funding;
