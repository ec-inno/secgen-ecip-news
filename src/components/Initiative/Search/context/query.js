import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
