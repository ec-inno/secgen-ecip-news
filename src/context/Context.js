import React from 'react';

const initialState = {
  hideOverlay: true,
};

export const SET_LANGUAGE_OVERLAY_VISIBILITY =
  'SET_LANGUAGE_OVERLAY_VISIBILITY';

const reducer = (state, action) => {
  switch (action.type) {
    case SET_LANGUAGE_OVERLAY_VISIBILITY: {
      return Object.assign({}, state, {
        hideOverlay: action.hideOverlay,
      });
    }

    default: {
      return initialState;
    }
  }
};

const Context = React.createContext();

const ContextProviderWithReducer = ({ children }) => {
  const [store, dispatch] = React.useReducer(reducer, initialState);

  return (
    <Context.Provider value={{ store, dispatch }}>{children}</Context.Provider>
  );
};

export { ContextProviderWithReducer, Context as default };
