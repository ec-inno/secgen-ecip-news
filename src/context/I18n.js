import React from 'react';

const I18n = React.createContext();

export default I18n;
