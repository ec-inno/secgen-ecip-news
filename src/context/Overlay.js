import React, { useContext } from 'react';

const Context = React.createContext();

const useOverlayContext = () => useContext(Context);

export { useOverlayContext, Context as default };
