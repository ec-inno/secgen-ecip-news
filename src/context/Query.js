import React, { useContext } from 'react';

import { defaultLangKey } from '../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'unsetFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

const useQueryContext = () => useContext(Context);

export { queryInit, queryReducer, Context as default, useQueryContext };
