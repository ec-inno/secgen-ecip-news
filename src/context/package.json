{
  "name": "@eci/context",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@ecl/eu-preset-website": "2.8.0",
    "react": "16.9.0"
  }
}
