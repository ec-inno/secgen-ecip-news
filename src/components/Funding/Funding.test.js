import React from 'react';
import renderer from 'react-test-renderer';

import * as I18nContext from '@eci/context/I18n';

import Funding from './Funding';

describe('Funding', () => {
  it('no input: returns an empty string', () => {
    const tree = renderer.create(<Funding />).toJSON();
    expect(tree).toBe('');
  });

  it('can render only lastUpdate', () => {
    const props = {
      funding: {
        lastUpdate: 'last update string',
      },
    };

    const tree = renderer.create(<Funding {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can render only totalAmount', () => {
    const props = {
      funding: {
        totalAmount: 100,
      },
    };

    const tree = renderer.create(<Funding {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does not render totalAmount when zero', () => {
    const props = {
      funding: {
        totalAmount: 0,
      },
    };

    const tree = renderer.create(<Funding {...props} />).toJSON();
    expect(tree).toBe('');
  });

  it('Renders a file section when document property is present', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
      location: {
        href: 'http://localhost:8000/en/initiatives/#1',
        pathname: '/en/initiatives/',
        hash: '#1',
      },
    }));

    const props = {
      funding: {
        document: {
          id: 16806,
          name: 'DummyMemberDoc1.txt',
          mimeType: 'text/plain',
          size: 10,
        },
      },
    };

    const tree = renderer.create(<Funding {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders a table when sponsors', () => {
    const props = {
      funding: {
        sponsors: [
          {
            name: 'foo',
            date: 'now',
            amount: 1,
          },
          {
            name: 'bar',
            date: 'then',
            amount: 2,
          },
          {
            name: 'baz',
            date: 'never',
            amount: 3,
          },
        ],
      },
    };

    const tree = renderer.create(<Funding {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
