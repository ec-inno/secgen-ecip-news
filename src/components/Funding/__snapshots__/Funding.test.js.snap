// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Funding Renders a file section when document property is present 1`] = `
Array [
  <h2
    className="ecl-u-type-heading-2"
  >
    Sources of support and funding
  </h2>,
  <p
    className="ecl-u-type-paragraph ecl-u-type-bold"
  >
    Further information on support and funding
    :
    <div
      className="ecl-file"
      data-ecl-file={true}
    >
      <div
        className="ecl-file__container"
      >
        <svg
          aria-hidden="true"
          className="ecl-file__icon ecl-icon ecl-icon--2xl ecl-icon ecl-icon--m"
          focusable="false"
        >
          
          
          <use
            xlinkHref="file#general--copy"
          />
        </svg>
        <div
          className="ecl-file__info"
        >
          <div
            className="ecl-file__title"
          >
            DummyMemberDoc1.txt
          </div>
          <div
            className="ecl-file__meta"
          >
            (
            10 Bytes
             -
             
            TXT
            )
          </div>
        </div>
        <a
          className="ecl-link ecl-link--standalone"
          download={true}
          href="undefined/register/document/16806"
        >
          <span
            className="ecl-link__label"
          >
            Download
          </span>
           
          <svg
            aria-hidden="true"
            className="ecl-icon--fluid ecl-link__icon ecl-icon ecl-icon--m"
            focusable="false"
          >
            
            
            <use
              xlinkHref="file#ui--download"
            />
          </svg>
        </a>
      </div>
    </div>
  </p>,
]
`;

exports[`Funding Renders a table when sponsors 1`] = `
Array [
  <h2
    className="ecl-u-type-heading-2"
  >
    Sources of support and funding
  </h2>,
  <table
    className="eci-table ecl-u-type-paragraph"
  >
    <tbody>
      <tr>
        <th>
          Name of sponsor
        </th>
        <th>
          Date
          <sup>
            [1]
          </sup>
        </th>
        <th>
          Amount in EUR
          <sup>
            [2]
          </sup>
        </th>
      </tr>
      <tr
        className=""
      >
        <td>
          foo
        </td>
        <td>
          now
        </td>
        <td>
          1
        </td>
      </tr>
      <tr
        className="dark-row"
      >
        <td>
          bar
        </td>
        <td>
          then
        </td>
        <td>
          2
        </td>
      </tr>
      <tr
        className=""
      >
        <td>
          baz
        </td>
        <td>
          never
        </td>
        <td>
          3
        </td>
      </tr>
    </tbody>
  </table>,
  <p
    className="ecl-u-type-paragraph-s"
  >
    [1]
     
    Date when the contribution was received. In case of in kind contributions, date when the in kind contribution started.
    <br />
    [2]
     
    Actual amount of funding provided or estimated value of in kind contribution.
  </p>,
]
`;

exports[`Funding can render only lastUpdate 1`] = `
Array [
  <h2
    className="ecl-u-type-heading-2"
  >
    Sources of support and funding
  </h2>,
  <p
    className="ecl-u-type-paragraph"
  >
    Last update
    : 
    last update string
  </p>,
]
`;

exports[`Funding can render only totalAmount 1`] = `
Array [
  <h2
    className="ecl-u-type-heading-2"
  >
    Sources of support and funding
  </h2>,
  <p
    className="ecl-u-type-paragraph ecl-u-type-bold"
  >
    Total amount of support and funding
    : 
    â‚¬
    100
  </p>,
]
`;
